### PR TITLE
policyengine: Remove ConnectivityPolicy

### PR DESCRIPTION
**Pull request by orozery**
This PR moves the policyengine to use v1alpha1.AccessPolicy instead of policytypes.ConnectivityPolicy.
The "matches" and "decides" functions are moved to connectivitypdp, as I think it is better to keep `pkg/apis` as short and simple as possible.

### CODE REVIEW

**Comment by zivnevo** [comment]
@orozery , can you please resolve conflicts?

File: pkg/policyengine/connectivitypdp/connectivity_pdp_test.go [hunk 1]
```diff
@@ -213,29 +252,18 @@ func addPoliciesFromFile(pdp *connectivitypdp.PDP, filename string) error {
...
-				}
-			case k8sshim.ConnectivityPolicyKind:
-				regPolicy := k8sshim.ConnectivityPolicy{ObjectMeta: metav1.ObjectMeta{Name: policy.Name}, Spec: policy.Spec}
-				err = pdp.AddOrUpdatePolicy(regPolicy.ToInternal())
-				if err != nil {
-					fmt.Printf("invalid connectivity policy: %v\n", err)
-				}
-			default: // TODO: log a warning
-				fmt.Printf("Object kind is not a connectivity policy: %s\n", policy.Kind)
+		var policy v1alpha1.AccessPolicy
```
Comment (zivnevo) [inline pkg/policyengine/connectivitypdp/connectivity_pdp_test.go:255]: documentation for `addPoliciesFromFile` needs an update

File: pkg/policyengine/connectivitypdp/connectivity_pdp_test.go [hunk 2]
```diff
@@ -26,53 +26,88 @@ import (
...
 
 var (
-	trivialLabel       = policytypes.WorkloadAttrs{"key": "val"}
+	trivialLabel       = connectivitypdp.WorkloadAttrs{"key": "val"}
 	trivialSelector    = metav1.LabelSelector{MatchLabels: trivialLabel}
-	trivialWorkloadSet = policytypes.WorkloadSetOrSelector{WorkloadSelector: &trivialSelector}
+	trivialWorkloadSet = v1alpha1.WorkloadSetOrSelector{WorkloadSelector: &trivialSelector}
 )
 
+func TestValidation(t *testing.T) {
```
Comment (zivnevo) [inline pkg/policyengine/connectivitypdp/connectivity_pdp_test.go:41]: This is a duplicate of the test in `accesspolicy_test.go`

**Review by zivnevo — CHANGES_REQUESTED** [review]

File: pkg/controlplane/authz/manager.go [hunk 3]
```diff
@@ -221,25 +222,7 @@ func (m *Manager) deleteAccessPolicy(_ types.NamespacedName) {
...
-	}
-
-	policyData, err := json.Marshal(&policytypes.ConnectivityPolicy{
-		Name:       accessPolicy.Name,
-		Privileged: accessPolicy.Spec.Privileged,
-		Action:     policytypes.PolicyAction(accessPolicy.Spec.Action),
-		From:       convert(accessPolicy.Spec.From),
-		To:         convert(accessPolicy.Spec.To),
-	})
+	policyData, err := json.Marshal(accessPolicy)
```
[the remark this answers is not in the document]
Comment (orozery) [reply pkg/controlplane/authz/manager.go:225]: yes, for LB

**Review by zivnevo — APPROVED** [review]

**Pull request merged**